lochness/transfer/__init__.py

```diff
@@ -0,0 +1,296 @@
+                           Lochness['phoenix_root'],
+                           tmpfilename.name,
+                           general_only)
+
+        # send to remote server
+        send_data_over_sftp(Lochness, tmpfilename.name)
+
+
+def lochness_to_lochness_transfer_receive(Lochness):
+    '''Get newly transferred file and decompress to PHOENIX
+
+    Key arguments:
+        Lochness: Lochness object loaded from the config.load, object.
+
+    Structure needed in the Lochness['keyring']
+        Lochness['keyring'] = {
+            'lochness_to_lochness_receive': {
+                'PATH_IN_HOST': '/SFTP/DATA/REPO'
+            }
+        }
+        Lochness['lochness_sync_history_csv']: '/SYNC/HISTORY.csv'
+
+    TODO: move PATH_IN_HOST to config?
+    '''
+
+    # sftp_keyring = Lochness['keyring']['lochness_to_lochness']
+    # host = sftp_keyring['HOST']
+    # path_in_host = sftp_keyring['PATH_IN_HOST']
+
+    sftp_keyring = Lochness['keyring']['lochness_to_lochness_receive']
+    path_in_host = sftp_keyring['PATH_IN_HOST']
+
+    target_phoenix_root = Lochness['phoenix_root']
+    sync_db = Lochness['lochness_sync_history_csv']
+
+    last_sync_timestamp, _, sync_df = get_ts_and_db(sync_db)
+
+    for root, _, files in os.walk(path_in_host):
+        for file in files:
+            file_p = Path(root) / file
+            # if the file is transferred after the last transfer
+            if file_p.stat().st_mtime > last_sync_timestamp \
+                    and file.endswith('.tar'):
+                decompress_transferred_file_and_copy(target_phoenix_root,
+                                                     file_p)
+
+    sync_df.to_csv(sync_db, index=False)
+
+
+def decompress_transferred_file_and_copy(target_phoenix_root: str,
+                                         tar_file_trasferred: str):
+    '''Decompress the tar file and arrange the new data into PHOENIX structure
+
+    Decompress the tar file transferred and arrange the new data into the
+    corresponding structure under the PHOENIX structure.
+
+    Key arguments:
+        target_phoenix_root: path of the PHOENIX directory, str.
+        tar_file_trasferred: path of the tar file transferred from another
+                             lochness using lochness_to_lochness_transfer, str.
+    '''
+    tar = tarfile.open(tar_file_trasferred)
+
+    with tf.TemporaryDirectory(suffix='tmp', dir='.') as tmpdir:
+        tar.extractall(path=tmpdir)
+
+        for root, _, files in os.walk(tmpdir):
+            for file in files:
+                relative_root = root.split(tmpdir + '/PHOENIX/')[1]
+                target_path = Path(target_phoenix_root) / relative_root / file
+                target_path.parent.mkdir(exist_ok=True, parents=True)
+
+                # this overwrites exsiting file
+                shutil.copy(Path(root) / file, target_path)
+
+                # permission change
+                os.chmod(target_path, 0o0644)
```

Comment:
Here is an actual suggestion, use `os.remove(tar_file_transferred)` at the end to save space.

Reply:
Thanks. I also wanted to bring this up to the team. Is it safe for the DPACC to remove the transferred tarball? I'd actually reduce a lot of anxiety from having the tarball saved somewhere, at least for a while.

Reply:
The tarfile removal added. We can discuss further about this later in a meeting with others.